operators/pkg/instautoctrl/terminator.go

```diff
@@ -179,7 +178,10 @@ func (r *InstanceTerminationReconciler) TerminateInstance(ctx context.Context, i
 	log := ctrl.LoggerFrom(ctx).WithName("termination")
 	log.Info("terminating instance")
 
-	instance.SetLabels(forge.InstanceAutomationLabelsOnTermination(instance.GetLabels()))
+	_, validityCheckErr := CheckEnvironmentValidity(ctx, r.Client, instance)
```

Comment:
Should distinguish between errors (the template does not exist) and the destination url is not set. In addition, you should log about both conditions.